sw/device/tests/penetrationtests/firmware/ibex_fi.c

```diff
@@ -702,6 +744,150 @@ status_t handle_ibex_fi_char_sram_write_static(ujson_t *uj) {
   return OK_STATUS();
 }
 
+status_t handle_ibex_fi_char_sram_write_read(ujson_t *uj)
+    __attribute__((optnone)) {
+  // Clear registered alerts in alert handler.
+  sca_registered_alerts_t reg_alerts = sca_get_triggered_alerts();
+
+  // Initialize SRAM region with inverse reference value.
+  sram_main_buffer[0] = ~ref_values[0];
+
+  // Init x5, x6, x6 with the reference values.
+  asm volatile("li x5, %0" : : "i"(ref_values[0]));
+  asm volatile("li x6, %0" : : "i"(ref_values[1]));
+  asm volatile("li x7, %0" : : "i"(ref_values[2]));
+
+  sca_set_trigger_high();
+  asm volatile(NOP10);
+  asm volatile("sw x5, (%0)" : : "r"((uint32_t *)&sram_main_buffer[0]));
+  asm volatile("lw x5, (%0)" : : "r"((uint32_t *)&sram_main_buffer[0]));
+  asm volatile("sw x6, (%0)" : : "r"((uint32_t *)&sram_main_buffer[0]));
+  asm volatile("lw x6, (%0)" : : "r"((uint32_t *)&sram_main_buffer[0]));
+  asm volatile("sw x7, (%0)" : : "r"((uint32_t *)&sram_main_buffer[0]));
+  asm volatile("lw x7, (%0)" : : "r"((uint32_t *)&sram_main_buffer[0]));
+  asm volatile("sw x5, (%0)" : : "r"((uint32_t *)&sram_main_buffer[0]));
```

Comment:
Why do you need the cast? sram_main_buffer is already an array of uint32_t's isn't it?
```suggestion
  asm volatile("sw x5, (%0)" : : "r"(&sram_main_buffer[0]));
``